proteus/TwoPhaseFlow/TwoPhaseFlowProblem.py

```diff
@@ -73,44 +86,106 @@ def __init__(self,
+        nd = self.nd
+        ns_model = self.ns_model
+        ls_model = self.ls_model
+        if boundaryConditions is not None:
             # check dirichlet BCs
-            assert 'pressure_increment_DBC' in boundaryConditions, "Provide pressure_increment_DBC"
+            if ns_model is not None:
+                assert 'pressure_DBC' in boundaryConditions, "Provide pressure_DBC"
+                assert 'vel_u_DBC' in boundaryConditions, "Provide vel_u_DBC"
+                assert 'vel_v_DBC' in boundaryConditions, "Provide vel_v_DBC"
+                if nd==3:
+                    assert 'vel_w_DBC' in boundaryConditions, "Provide vel_w_DBC"
+            if ls_model == 0:
+                assert 'vof_DBC' in boundaryConditions, "Provide vof_DBC"
+                assert 'ncls_DBC' in boundaryConditions, "Provide ncls_DBC"
+            elif ls_model == 1:
+                assert 'clsvof_DBC' in boundaryConditions, "Provide clsvof_DBC"
             # check advective flux BCs
-            assert 'pressure_increment_AFBC' in boundaryConditions,"Provide pressure_increment_AFBC"
+            if ns_model is not None:
+                assert 'pressure_AFBC' in boundaryConditions, "Provide pressure_AFBC"
+                assert 'vel_u_AFBC' in boundaryConditions, "Provide vel_u_AFBC"
+                assert 'vel_v_AFBC' in boundaryConditions, "Provide vel_v_AFBC"
+                if nd==3:
+                    assert 'vel_w_AFBC' in boundaryConditions, "Provide vel_w_AFBC"
+            if ls_model == 0:
+                assert 'clsvof_AFBC' in boundaryConditions, "Provide clsvof_AFBC"
+            if ls_model == 0:
+                assert 'vof_AFBC' in boundaryConditions, "Provide vof_AFBC"
             # check diffusive flux BCs
-            assert 'pressure_increment_DFBC' in boundaryConditions,"Provide pressure_increment_DFBC"
+            if ns_model is not None:
+                assert 'vel_u_DFBC' in boundaryConditions, "provide vel_u_DFBC"
+                assert 'vel_v_DFBC' in boundaryConditions, "provide vel_v_DFBC"
+                if nd==3:
+                    assert 'vel_w_DFBC' in boundaryConditions, "provide vel_w_DFBC"
+            if ls_model == 1:
+                assert 'clsvof_DFBC' in boundaryConditions, "provide clsvof_DFBC"
+            if ns_model==1: #rans3p
+                # check dirichlet BCs
+                assert 'pressure_increment_DBC' in boundaryConditions, "Provide pressure_increment_DBC"
+                # check advective flux BCs
+                assert 'pressure_increment_AFBC' in boundaryConditions,"Provide pressure_increment_AFBC"
+                # check diffusive flux BCs
+                assert 'pressure_increment_DFBC' in boundaryConditions,"Provide pressure_increment_DFBC"
+        else:
+            assert self.domain.useSpatialTools is not None, 'Either define boundaryConditions dict or use proteus.mprans.SpatialTools to set Boundary Conditions and run function assembleDomain'
```

Comment:
I think this is what you meant before, right? Basically to allow the user not to define `boundaryConditions` and assert that `SpatialTools` is being used.

Reply:
Yes, if SpatialTools was used, then BoundaryConditions associated to the geometry boundaries will be used too. I will add a switch within TwoPhaseFlowProblem in case someone wants to use SpatialTools for the geometry but still the old way of defining BC